### ISSUE TITLE
ArgumentException: An item with the same key has already been added. Key: RetryCount

### ISSUE THREAD

**Issue by DaveSweeton** [HangfireIO/Hangfire#1741]
On trying to view the details for a job that failed through the HangFire dashboard:

```
System.ArgumentException: An item with the same key has already been added. Key: RetryCount
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](List`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector)
   at Hangfire.SqlServer.SqlServerMonitoringApi.<>c__DisplayClass21_0.<JobDetails>b__0(DbConnection connection)
   at Hangfire.SqlServer.SqlServerStorage.UseConnection[T](DbConnection dedicatedConnection, Func`2 func)
   at Hangfire.SqlServer.SqlServerMonitoringApi.JobDetails(String jobId)
   at Hangfire.Dashboard.Pages.JobDetailsPage.Execute()
   at Hangfire.Dashboard.RazorPage.TransformText(String body)
   at Hangfire.Dashboard.RazorPageDispatcher.Dispatch(DashboardContext context)
   at Hangfire.Dashboard.AspNetCoreDashboardMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
```

**Comment by DaveSweeton** [comment]
This was using v1.7.13

**Comment by odinserj** [comment]
Hm, interesting. Could you show me the output of the following query issued against your SQL Server and ensure that PK_HangFire_JobParameter index is unique?

```sql
SELECT * FROM [HangFire].[JobParameter] WHERE [JobId] = <jobid>
```

![Uploading image.png…]()

**Comment by odinserj** [comment]
Hm, now I see that it's possible you are using older schema that contains only `IX_HangFire_JobParameter_JobIdAndName` index that wasn't unique. So I'm releasing 1.7.14 today with rolled back changes for query for `SetJobParameter` method introduced in 1.7.13. But you'll need to manually delete duplicates in the JobParameter table.

**Comment by odinserj** [comment]
Fixed in bf930462765922a1e22f1922c93773f03abbfa23, thank you so much for reporting this!
